### ISSUE TITLE
Errors while trying to start from scratch?

### ISSUE THREAD

**Issue by chief7** [gooofy/zamia-ai#5]
Hi @gooofy ,

I have tried to use your scripts for quite a long time now - and I did have some success, so thank you first of all for sharing this code with us.

Anyway - I keep getting errors while training the `nnet3_tdnn_a` (I didn't try any LSTM yet) and those errors seem to be caused by the fact that the utterance database used to train the network is already built with all the utterances you use to train your models and that not all of the data you use is being downloaded by your scripts.

Is there any way to start from scratch with a clean model just using the utterances we've downloaded this time?

Thanks in advance!

**Comment by gooofy** [comment]
First of all: sorry for my late response. I was in the middle of a big code refactoring when this issue came up and decided to finish the refactoring first and answer once it is complete - which it is, as of just now :) Please note that the speech/voxforge-related code moved to a separate repository "speech" now - the code has undergone quite a bit of cleanup so it might be worth trying but I have not addressed any issues of starting from scratch yet.

While I have no specific instructions or code ready I can say that I see two ways forward here: we could either try to make the scripts more robust against missing data or we could look into what data is missing and upload it to voxforge or other server. AFAIK all audio data I am using to build my models is free, however uploading bulk data to voxforge via FTP is a semi-manual process which is a bit error-prone, I guess that is the main reason you're missing data in your builds.

Let me know what you think.

**Comment by chief7** [comment]
Hi, first of all: It's really nice to see all the progress with your projects! Congratulations on that! Has been a huge inspiration for the ones I try to build myself. I get the feeling we're working on kind of the same stuff.

To answer your question: I guess it's more useful for us to have the scripts some kind of fail safe concerning missing data. Of course it would be nice to have all the data you're using to build models which are as accurate as yours are but most probably errors will come up now and then so I'd value the more robust scripts more.

Anyway, I'm willing to help if need a hand. Just let me know if that's ok for you!

Thanks again!

**Comment by gooofy** [comment]
I will close this issue as the speech related scripts have been moved to their separate repository now and we have a new ticket there.
